.github/workflows/build-ampd-and-push-to-r2.yaml

```diff
@@ -0,0 +1,61 @@
+name: Upload ampd contracts to Cloudflare R2 bucket
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: Github branch to checkout for compilation
+        required: true
+        default: main
+        type: string
+
+
+jobs:
+  compile-and-upload:
+    name: Compile contracts and push to R2
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+      packages: write
+      id-token: write
+    steps:
+      - name: Determine branch
+        id: get-branch-name
+        run: |
+          if [ "${{ github.event_name }}" == "push" ]; then
+            branch="main"
+          else
+            branch="${{ inputs.branch }}"
+          fi
+          echo "branch=$branch" >> $GITHUB_OUTPUT
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: "0"
+          path: axelar-amplifier
+          submodules: recursive
+          ref: ${{ steps.get-branch-name.outputs.branch }}
+
+      - name: Compile amplifier contracts
+        id: compile-contracts
+        run: |
+          cd axelar-amplifier
+          docker run --rm -v "$(pwd)":/code --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry cosmwasm/rust-optimizer:0.15.1
```

Comment:
any specific reason this is our approach as opposed to cargo build in a linux machine ?

Reply:
following existing amplifier compilation method in depoloy-devnet files:
https://github.com/axelarnetwork/axelarate/blob/main/.github/workflows/reuseable-deploy-devnet.yaml#L222
Also, the use of `rust-optimizer` that `cargo build` does not handle on itself I guess.

Reply:
@cgorenflo Can you also comment please?

Reply:
the `cosmwasm/rust-optimizer` image compiles and optimizes the contract binaries in a reproducible way. It's the recommended approach for building cosmwasm contracts